server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -967,20 +973,19 @@ private static void updateRemoteStoreSettings(Settings.Builder settingsBuilder,
                 return;
             }
 
-            // Verify REPLICATION_TYPE cluster level setting is not conflicting with Remote Store
-            if (INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings) == false
-                && CLUSTER_REPLICATION_TYPE_SETTING.get(clusterSettings).equals(ReplicationType.DOCUMENT)) {
+            // Verify index has replication type as SEGMENT
+            if (ReplicationType.DOCUMENT.toString().equals(settingsBuilder.get(SETTING_REPLICATION_TYPE))) {
```

Comment:
I think the null case here should also throw.

`if (ReplicationType.SEGMENT.toString().equals(settingsBuilder.get(SETTING_REPLICATION_TYPE)) == false) {`